tests/installation/bootloader_s390.pm

```diff
@@ -0,0 +1,427 @@
+use base "installbasetest";
+
+use testapi;
+
+use strict;
+use warnings;
+use English;
+
+use Data::Dumper qw(Dumper);
+use Carp qw(confess cluck carp croak);
+
+use feature qw/say/;
+
+###################################################################
+# linuxrc helpers
+
+sub linuxrc_menu() {
+    my ($self, $menu_title, $menu_entry) = @_;
+    # get the menu (ends with /^>/)
+    my $r = $self->{s3270}->expect_3270(output_delim => qr/^> /);
+    ### say Dumper $r;
+
+    # newline separate list of strings when interpolating...
+    local $LIST_SEPARATOR = "\n";
+
+    if (!grep /^$menu_title/, @$r) {
+        confess "menu does not match expected menu title ${menu_title}\n @${r}";
+    }
+
+    my @match_entry = grep /\) $menu_entry/, @$r;
+
+    if (!@match_entry) {
+        confess "menu does not contain expected menu entry ${menu_entry}:\n@${r}";
+    }
+
+    my ($match_id) = $match_entry[0] =~ /(\d+)\)/;
+
+    my $sequence = ["Clear", "String($match_id)", "ENTER"];
+
+    $self->{s3270}->sequence_3270(@$sequence);
+}
+
+sub linuxrc_prompt () {
+    my ($self, $prompt, %arg) = @_;
+
+    $arg{value}   //= '';
+    $arg{timeout} //= 3;
+
+    my $r = $self->{s3270}->expect_3270(output_delim => qr/(?:\[.*?\])?> /, timeout => $arg{timeout});
+
+    ### say Dumper $r;
+
+    # two lines or more
+    # [previous repsonse]
+    # PROMPT
+    # [more PROMPT]
+    # [\[EXPECTED_RESPONSE\]]>
+
+    # newline separate list of strings when interpolating...
+    local $LIST_SEPARATOR = "\n";
+
+    if (!grep /^$prompt/, @$r[0..(@$r-1)] ) {
+        confess"prompt does not match expected prompt (${prompt}) :\n"."@$r";
+    }
+
+    my $sequence = ["Clear", "String($arg{value})", "ENTER"];
+    push @$sequence, "ENTER" if $arg{value} eq '';
+
+    $self->{s3270}->sequence_3270(@$sequence);
+
+}
+
+
+sub ftpboot_menu () {
+    my ($self, $menu_entry) = @_;
+    # helper vars
+    my ($r_screenshot, $r_home_position, $s_home_position, $cursor_row, $row);
+
+    $r_screenshot = $self->{s3270}->expect_3270(clear_buffer => 1, flush_lines => undef, buffer_ready => qr/PF3=QUIT/);
+
+    # choose server
+
+    $r_home_position = $self->{s3270}->send_3270("Home");
+    # Perl question:
+    # Why can't I just call this function?  why do I need & ??
+    # and why this FQDN?
+    $s_home_position = &backend::s390x::s3270::nice_3270_status($r_home_position->{terminal_status});
+
+    $cursor_row = $s_home_position->{cursor_row};
+
+    ## say Dumper @$r_screenshot;
+
+    while ( ($row, my $content) = each(@$r_screenshot)) {
+        if ($content =~ $menu_entry) {
+            last;
+        }
+    }
+
+    my $sequence = ["Home", ("Down") x ($row-$cursor_row), "ENTER", "Wait(InputField)"];
+    ### say "\$sequence=@$sequence";
+
+    $self->{s3270}->sequence_3270(@$sequence);
+
+    return $r_screenshot;
+}
+
+###################################################################
+require Text::Wrap;
+
```

Comment:
I don't understand.
The warnings in the test suite come from the Test::Wrap API which works with setting variables, which is ugly but is what it is.
or are you suggesting to reimplement Text::Wrap ??